### PR TITLE
Cleaned up all the warn-unused and warn-unused-import issues

### PR DESCRIPTION
**Pull request by norbert-radyk**
The change is a pure cleanup change (no functional changes) to fix the problems reported by the unused checks (-Ywarn-unused and -Ywarn-unused-import) and enforcing the checks by adding -Xfatal-warnings (will result in a compilation error should any new unused code section be introduced). 

In some cases (where the method is a public one and can be overriden by the client) in order not to break the API removing the warning required marking the parameter explicitly as unused based on a solution proposed here: https://stackoverflow.com/questions/46128561/excluding-type-evidence-parameters-from-analysis-in-scala-when-using-ywarn-unus

### CODE REVIEW

File: src/main/scala/org/squeryl/logging/BarChartRenderer.scala [hunk 1]
```diff
@@ -94,8 +93,9 @@ object BarChartRenderer {
     sb.toString
    }
 
-
-  def page(stats: Stat*) = """
+  def page(stats: Stat*) = {
+    stats.unused
+    """
```
Comment (xuwei-k) [inline src/main/scala/org/squeryl/logging/BarChartRenderer.scala]: Maybe missing `s` interpolation 🤔 ?

https://github.com/squeryl/squeryl/blob/7b2b1703f50aee8cb2772189c264b03fc237c5ff/src/main/scala/org/squeryl/logging/BarChartRenderer.scala#L111-L113

```diff
          function drawVisualization() {{
-            {Unparsed(funcCalls(stats))}
+            ${Unparsed(funcCalls(stats))}
          }}
```

File: src/main/scala/org/squeryl/package.scala [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org
+
+package object squeryl {
+
+  implicit class IdOps[A](a: A) {
```
Comment (xuwei-k) [inline src/main/scala/org/squeryl/package.scala]: I think `"-Xfatal-warnings"` and `IdOps` are too much 😕

**Comment by xuwei-k** [comment]
Hmmm 🤔 I think some (not private method) parameters (e.g. `ev: CanCompare[T1, T2]` ) has meaning. Don't remove these params.
Could you please separate more small pull requests? (e.g. "remove unused imports", "add return type annotations", "remove unnecessary comments")

**Comment by norbert-radyk** [comment]
Thanks @xuwei-k for the feedback. Having separate merge requests for the smaller sets of changes is a fair point. I am going to close this one and have raised a small one for the fix on the BarChartRenderer which was failing with the unused import: https://github.com/squeryl/squeryl/pull/248

**Pull request closed without merging**